pkg/labels/http_applicator.go

```diff
@@ -0,0 +1,84 @@
+package labels
+
+import (
+	"encoding/json"
+	"net/http"
+	"net/url"
+
+	"github.com/square/p2/pkg/logging"
+	"github.com/square/p2/pkg/util"
+)
+
+type httpApplicator struct {
+	client *http.Client
+	// The endpoint that will be queried for matches.
+	// GetMatches will add to this endpoint a query parameter with key "selector" and value selector.String()
+	matchesEndpoint *url.URL
+	logger          logging.Logger
+}
+
+var _ Applicator = &httpApplicator{}
+
+func NewHttpApplicator(client *http.Client, matchesEndpoint *url.URL) (*httpApplicator, error) {
+	if matchesEndpoint == nil {
+		return nil, util.Errorf("matches endpoint cannot be nil")
+	}
+
+	c := client
+	if c == nil {
+		c = http.DefaultClient
+	}
+
+	return &httpApplicator{
+		logger:          logging.DefaultLogger,
+		matchesEndpoint: matchesEndpoint,
+		client:          c,
+	}, nil
+}
+
+func (h *httpApplicator) SetLabel(labelType Type, id, name, value string) error {
+	return util.Errorf("SetLabel not implemented for HttpApplicator (type %s, id %s, name %s, value %s)", labelType, id, name, value)
+}
+
+func (h *httpApplicator) RemoveLabel(labelType Type, id, name string) error {
+	return util.Errorf("RemoveLabel not implemented for HttpApplicator (type %s, id %s, name %s)", labelType, id, name)
+}
+
+func (h *httpApplicator) GetLabels(labelType Type, id string) (Labeled, error) {
+	return Labeled{}, util.Errorf("GetLabels not implemented for HttpApplicator (type %s, id %s)", labelType, id)
+}
+
+func (h *httpApplicator) GetMatches(selector Selector, labelType Type) ([]Labeled, error) {
```

Comment:
well right now the backend only supports type=NODE, therefore the implementation only has an endpoint for type=NODE. adding support for type=POD to the backend would be fairly straightforward in which case same could be implemented here. for now I don't think it's a blocker?